examples/distributed/pyg/distributed_cpu.py

```diff
@@ -0,0 +1,497 @@
+        concurrency=concurrency,
+        async_sampling=async_sampling,
+    )
+
+    print('--- Initialize model ...')
+    # Define model and optimizer.
+    model = GraphSAGE(
+        in_channels=128 if is_hetero else 100,  # num_features
+        hidden_channels=256,
+        num_layers=len(num_neighbors),
+        out_channels=349 if is_hetero else 47,  # num_classes in dataset
+    ).to(current_device)
+
+    @torch.no_grad()
+    def init_params():
+        # Init parameters via forwarding a single batch to the model:
+        print('--- Initialize parameters of the hetero model ...')
+        with train_loader as iterator:
+            batch = next(iter(iterator))
+            batch = batch.to(current_device, 'edge_index')
+            model(batch.x_dict, batch.edge_index_dict)
+            del batch
+            torch.distributed.barrier()
+
+    if is_hetero:
+        # Turn model to hetero and initialize parameters
+        metadata = [meta['node_types'], [tuple(e) for e in meta['edge_types']]]
+        model = to_hetero(model, metadata).to(current_device)
+        init_params()
+        torch.distributed.barrier()
+
+    # Enable DDP
+    model = DistributedDataParallel(model, find_unused_parameters=is_hetero)
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.0004)
+    torch.distributed.barrier()
+
+    # Train and test
+    print(f'--- Start training for {num_epochs} epochs ...')
+    for i in range(num_epochs):
+        start = time.time()
+        epoch = i + 1
+        print(f'Train Epoch {epoch}/{num_epochs}')
+        model.train()
+        loss = training(
+            is_hetero,
+            model,
+            train_loader,
+            optimizer,
+            current_ctx,
+            current_device,
+            epoch,
+            logfile,
+            num_loader_threads,
+            progress_bar,
+        )
+        print(f'[Node {current_ctx.rank}] Epoch {epoch}: \
+                Train Loss = {loss:.4f}, \
+                Train Time = {(time.time() - start):.2f}')
+
+        # Test accuracy.
+        if i % 5 == 0:
+            start = time.time()
+            print(f'Test Epoch {epoch}/{num_epochs}')
+            model.eval()
+            acc = test(
+                model,
+                test_loader,
+                is_hetero,
+                current_ctx,
+                current_device,
+                epoch,
+                logfile,
+                num_loader_threads,
+                progress_bar,
+            )
+            print(f'[Node {current_ctx.rank}] Epoch {epoch}: '
+                  f'Test Accuracy = {acc:.4f}, '
+                  f'Test Time = {(time.time() - start):.2f}')
+    print(f'--- [Node {current_ctx.rank}] Closing ---')
+    torch.distributed.destroy_process_group()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Arguments for distributed training of supervised SAGE.')
+    parser.add_argument(
+        '--dataset',
```

Comment:
If you add the following restriction in here: choices=['ogbn-products', ognb-mag']
Then you can remove everything between lines 161-167.

Reply:
fixed